### ISSUE TITLE
[ENH] AptaDiff: Noise, denoise schedules, loss functions, and transformer layers

### ISSUE THREAD

**Issue by aditi-dsi** [gc-os-ai/pyaptamer#738]
We should implement the core tensor operations for the AptaDiff model, keeping the logic separated from the main estimator.

The goal is to add stateless PyTorch functions to handle the mathematical logic for:
- **Schedules:** Generating the noise arrays.
- **Noise & Denoise Steps:** Calculating how to add noise to the sequences (forward process) and how the model predicts the denoised steps (reverse process).
- **Loss Functions:** Calculating the training loss (ELBO), combining the KL divergence and cross-entropy reconstruction loss.

For layers, we should add:
- **Transformer Layers:** Positional Embedding and Linear Attention Transformer embedding layers. We should vendorize the original transformers implementation being used as the backend of the aforementioned two layers from the original author's [repository](https://github.com/lucidrains).

**Comment by AIMindCrafter** [comment]
i wanna contribute in this
